walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,65 @@
 
 package com.google.sps;
 
+import java.util.Comparator;
+import java.util.List;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+
 
 public final class FindMeetingQuery {
+
+ /* Given the meeting information, this method
+    returns the times when the meeting could happen that day.
+ */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> options = new ArrayList<TimeRange>();
+    if (Integer.compare((int) request.getDuration(), TimeRange.WHOLE_DAY.duration()) > 0) {
+        // If duration of event is more than 24hrs
+        return options;
+    }
+
+    if (events.equals(Collections.emptySet())) {
+        // If there are no events scheduled
+        options.add(TimeRange.WHOLE_DAY);
+        return options;
+    }
+    
+    // Most recent start time for the chunk of meeting time option
+    int currentStart = 0;
+
+    List<Event> eventsList = new ArrayList<Event>(events);
+    // eventsList.sort(Comparator.comparing(Event::getWhen));
+    // Iterate through event list in order
+    System.out.println("eventsList.size(): " + eventsList.size());
+    for (int i = 0; i < eventsList.size(); i++) {
```

Comment:
Might be better to use a forEach loop, so that you don't have to call `eventsList.get(i)` everytime.

`for (Event e : eventsList) { `